cronjob_scripts/crates_io_popular_crates.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import tarfile
+import warnings
+from pathlib import Path
+
+import requests
+import polars as pl
+
+warnings.filterwarnings("ignore", message="Polars found a filename")
+
+class TarEntry:
+    def __init__(self, tarball: tarfile.TarFile, member: tarfile.TarInfo):
+        self.tarball = tarball
+        self.member = member
+
+    def is_one_of_csvs_interested(self, csvs_to_extract: tuple[str, ...]) -> str | None:
+        if not self.member.isfile():
+            return None
+        for name in csvs_to_extract:
+            if self.member.name.endswith(f"data/{name}"):
+                return name
+        return None
+
+    def get_file_stream(self):
+        stream = self.tarball.extractfile(self.member)
+        assert stream
+        return stream
+
+def download_tar_gz(url):
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+
+    with tarfile.open(fileobj=response.raw, mode='r:gz') as tarball:
+        while True:
+            member = tarball.next()
+            if member is None:
+                return
+            yield TarEntry(tarball, member)
+
+def get_crates_io_popular_crates_inner(minimum_downloads=200000):
+    files_to_extract = ("crate_downloads.csv", "crates.csv", "default_versions.csv", "versions.csv")
```

Comment:
```suggestion
    files_to_extract = ["crate_downloads.csv", "crates.csv", "default_versions.csv", "versions.csv"]
``` 

use list rather than tuple if you have uniform types and the length is not relevant - you would use &[&str] in rust, so list in python

Reply:
Thanks updated